src/components/dropZone.tsx

```diff
@@ -1,63 +1,93 @@
+};
 
 export default function DropzoneFiles() {
-  const parsingModelFilepath = Object.keys(parsingModel.parsingmodel);
-  const [files] = useState('');
-  const handleChange = (e: any) => {
-    var fileReader = new FileReader();
-    // The FileReader object lets web applications asynchronously read the contents of files (or raw data buffers) stored on the user's computer, using File or Blob objects to specify the file or data to read.
-    fileReader.readAsArrayBuffer(e.target.files[0]);
-    // Method used to start reading the contents of a specified Blob or File. When the read operation is finished, the readyState becomes DONE, and the loadend is triggered. At that time, the result attribute contains an ArrayBuffer representing the file's data.
-    var smartDataBloc = [] as any;
-    // Create list to store the smartData (smartParser's output)
-    fileReader.onload = (e: any) => {
-      // The FileReader.onload property contains an event handler executed when the FileReader.load_event event is fired, when content read with readAsArrayBuffer, readAsBinaryString, readAsDataURL or readAsText is available.
-      JSZip.loadAsync(e.target.result).then(function (zip: any) {
-        Object.keys(zip.files).forEach(function (filename) {
-          // The Object.keys() method returns a table with all the properties of the unzip file. it follows an iteration
-          if (filename.split('.')[1] === 'json') {
-            for (let i = 0; i < parsingModelFilepath.length; i++) {
-              if (filename.endsWith(parsingModelFilepath[i])) {
-                zip.files[filename]
-                  .async('string')
-                  .then(function (fileData: any) {
-                    const fileContent = JSON.parse(fileData); // convert the type of fileData variable from String to Object
-                    if (Object.keys(fileContent).length > 0) {
-                      var newElement = smartParser(filename, fileContent) as any; // execute parsing function
-                      smartDataBloc.push(... newElement);
-                    } else {
-                      console.warn('The file is empty');
-                    }
-                  });
-              }
-            }
-          } else if (filename.split('.')[1] === 'csv') {
-            console.log('csv file [WIP]');
-          } else if (filename.split('.').length > 1) {
-            console.log('unknown file type');
+  const parsingModelFilepath = Object.keys(parsingModel);
+
+  const [smartDataBloc, setSmartDataBloc] = useState<string[][]>([]);
+
+  const { register, handleSubmit } = useForm<FormType>(); // initialize the hook
+
+  const onSubmit = async (data: FormType) => {
+    const preparedData = await prepareData(data);
+
+    for (const [filename, fileproperties] of Object.entries(
+      preparedData.files
+    )) {
+      if (filename.split('.')[1] === 'json') {
+        // semi-risky way to guess if it's a json file; maybe we should combine it with a regexp
```

Comment:
Une solution comme [celle-ci ](https://stackoverflow.com/questions/43669690/how-to-validate-json-file-extension-with-regular-expression)serait donc plus robuste ? Qu'est-ce que tu en penses ?

Reply:
Oui ça serait déjà mieux :)